File: tooldemo/src/main/java/com/excellence/tooldemo/ViewPagerAdapterActivity.java
package com.excellence.tooldemo;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.excellence.basetoolslibrary.pageradapter.BasePagerAdapter;

public class ViewPagerAdapterActivity extends AppCompatActivity {
    
    private ViewPager mViewPager = null;
    private NumAdapter mAdapter = null;
    private int mPageCount = 1;
    private int mNumScale = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view_pager_adapter);

        mViewPager = (ViewPager) findViewById(R.id.view_pager);
        mAdapter = new NumAdapter(mPageCount);
        mViewPager.setAdapter(mAdapter);
    }

    private class NumAdapter extends BasePagerAdapter {

        public NumAdapter(int pageCount) {
            super(pageCount);
        }

        @Override
        protected View loadView(Context context, int pageIndex) {
            // 加载每页
            TextView textView = new TextView(context);
            textView.setText(String.valueOf(pageIndex * mNumScale));
            textView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mNumScale *= 2;
                    mPageCount += 1;
                    // 刷新每页
                    mAdapter.notifyNewData(mPageCount);
                }
            });
            return textView;
        }
    }
}

File: tooldemo/src/main/java/com/excellence/tooldemo/ViewPagerAdapterActivity.kt
package com.excellence.tooldemo

import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.excellence.basetoolslibrary.pageradapter.BasePagerAdapter

class ViewPagerAdapterActivity : AppCompatActivity() {

    private var mViewPager: ViewPager? = null
    private var mAdapter: NumAdapter? = null
    private var mPageCount = 1
    private var mNumScale = 1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_view_pager_adapter)
        mViewPager = findViewById<View>(R.id.view_pager) as ViewPager
        mAdapter = NumAdapter(mPageCount)
        mViewPager!!.adapter = mAdapter
    }

    private inner class NumAdapter(pageCount: Int) : BasePagerAdapter(pageCount) {
        override fun loadView(context: Context, pageIndex: Int): View {
            // 加载每页
            val textView = TextView(context)
            textView.text = (pageIndex * mNumScale).toString()
            textView.setOnClickListener {
                mNumScale *= 2
                mPageCount += 1
                // 刷新每页
                mAdapter!!.notifyNewData(mPageCount)
            }
            return textView
        }
    }
}